docs/content/preview/architecture/transactions/distributed-txns.md

```diff
@@ -85,3 +85,17 @@ After a transaction is committed, the following two fields are set:
 
 - Commit hybrid timestamp: This timestamp is chosen as the current hybrid time at the transaction status tablet at the moment of appending the transaction committed entry to its Raft log. It is then used as the final MVCC timestamp for regular records that replace the transaction's provisional records when provisional records are being applied and cleaned up.
 - List of IDs of participating tablets: After a transaction commits, the final set of tablets to which the transaction has written is known. The tablet server managing the transaction sends this list to the transaction status tablet as part of the commit message, and the transaction status tablet makes sure that all participating tablets are notified of the transaction's committed status. This process might take multiple retries, and the transaction record can only be cleaned up after this is done.
+
+## Impact of Failures
+
+The Provisional Records are written to all the replicas of the tablets responsible for the keys being modified in a transaction. When a node with the tablet that has received the provisional records or is about to receive the provisional records fails, a new leader is elected for the tablet within a few seconds(`~2s`) as described in [Leader Failure](../../core-functions/high-availability/#tablet-peer-leader-failure). The transaction simply proceeds further with the newly elected leader. In this case, the time taken for the transaction to complete increases by the time taken for the leader election.
+
+When the transaction manager (typically, the node the client is connected to) does not receive a heartbeat from the client, it automatically cancels that transaction, so those provisional records would no longer block conflicting transactions waiting on the same keys. The transaction manager also maintains the transaction-id to client mapping and sends heartbeats to the transaction status tablet containing the information about the the transaction. When the manager fails, the heartbeats stop and the provisional records remain uncommitted in the respective tablets. Transactions that do not get heartbeats at the transaction status tablet become stale after certain time. Clients connected to the failed manager will receive an error message: 
```

Comment:
in this case i think its actually the coordinator who cancels the transaction if it doesn't receive a heartbeat from the pg client

> The transaction manager also maintains the transaction-id to client mapping and sends heartbeats to the transaction status tablet containing the information about the the transaction

i'm not sure what this sentence is trying to say, can you rephrase?

> When the manager fails, the heartbeats stop and the provisional records remain uncommitted in the respective tablets.

why would the manager fail? give an example

> Transactions that do not get heartbeats at the transaction status tablet become stale after certain time.

this is the same case as above, so maybe note that. also "stale after certain time" is imprecise -- we can explain how that happens in terms of these flags: https://github.com/yugabyte/yugabyte-db/blob/master/src/yb/tablet/transaction_coordinator.cc#L74

Reply:
Explained offline and Made appropriate changes.